### ISSUE TITLE
PandasJsonCacher errors with maximum recursion level reached

### ISSUE THREAD

**Issue by stewartsl** [ORNL/curifactory#3]
**Describe the bug**
PandasJsonCacher errors with
```
ERROR - OverflowError - Maximum recursion level reached
```
The PandasCsvCacher works with this same object

**To Reproduce**
1. Use the PandasJsonCacher as a stage cacher for an output with at least 17 columns and 5 rows
2. experiment will error out on that stage
3. Replace the cacher  with the PandasCsvCacher
4. Caching will work as expected

Note that in this case, exif data from images is being stored in the dataframe to be cached

**Expected behavior**
Expect Json Cached object to be created

**Comment by WildfireXIII** [comment]
Similar to #4, I think this might be specific to the exif data you're using. The cacher works for random numpy data with that shape, and I've added a test case for that to be sure. Also similar to #4 I've exposed an optional dictionary to pass to `to_json` if there's any parameters there that might solve the problem.

I suspect using a `PickleCacher` (https://ornl.github.io/curifactory/latest/caching.html#curifactory.caching.PickleCacher) is probably your best bet for data in this format.
